### PR TITLE
[Dashboard filters] Create initial infrastructure and group repros

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
As part of an ongoing effort defined in https://github.com/metabase/metabase/issues/17078, this is the first PR that sets the ground for more tests.
- creates new e2e CI group
- moves over some of the existing dashboard filter tests
- groups dashboard filter reproductions together under the same folder and separates them by one repro per file

### CODE REVIEW

File: frontend/test/metabase/scenarios/dashboard-filters/reproductions/12720.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+        cy.request("POST", "/api/dashboard/1/cards", {
+          cardId: SQL_ID,
+        }).then(({ body: { id: SQL_DASH_CARD_ID } }) => {
+          cy.log(
+            "Edit both cards (adjust their size and connect them to the filter)",
+          );
+
+          cy.request("PUT", "/api/dashboard/1/cards", {
+            cards: [
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/dashboard-filters/reproductions/12720.cy.spec.js:57]: I'd consider extracting the setting of this object to a function.

If I understood it correctly, we'd pass `dashboardFilter.id` as the argument and get back `{ cards: [……]}`

Same principle as setting the object consts above in this file.

File: frontend/test/metabase/scenarios/dashboard-filters/reproductions/13150.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        cy.log("Add 3 filters to the dashboard");
+        cy.request("PUT", `/api/dashboard/${dashboard_id}`, {
+          parameters,
+        });
+
+        cy.log("Add previously created qeustion to the dashboard");
+
+        cy.log("Connect all filters to the card");
+        cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+          cards: [
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/dashboard-filters/reproductions/13150.cy.spec.js:45]: I'd also consider extracting this to a function.

It'd have 5 arguments if I am not mistaken, which is a bit too much. 
All in all this test might still be easier to step into.

File: frontend/test/metabase/scenarios/dashboard-filters/reproductions/15119.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      type: "category",
+    };
+
+    cy.createQuestionAndDashboard({ questionDetails }).then(
+      ({ body: { id, card_id, dashboard_id } }) => {
+        cy.addFilterToDashboard({ filter, dashboard_id });
+
+        // Connect filter to the card
+        cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+          cards: [
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/dashboard-filters/reproductions/15119.cy.spec.js:31]: Same suggestion as above of a possible function.

File: frontend/test/metabase/scenarios/dashboard-filters/reproductions/15279.cy.spec.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    cy.createQuestionAndDashboard({ questionDetails }).then(
+      ({ body: { id, card_id, dashboard_id } }) => {
+        // Add filters to the dashboard
+        cy.request("PUT", `/api/dashboard/${dashboard_id}`, {
+          parameters,
+        });
+
+        // Connect filters to that question
+        cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+          cards: [
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/dashboard-filters/reproductions/15279.cy.spec.js:51]: Same suggestion as above of a possible function.

File: frontend/test/metabase/scenarios/dashboard-filters/reproductions/15689.cy.spec.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  });
+
+  it("dashboard filters should allow multiple selections (metabase#15689)", () => {
+    cy.createQuestionAndDashboard({ questionDetails }).then(
+      ({ body: { id, card_id, dashboard_id } }) => {
+        cy.addFilterToDashboard({ filter, dashboard_id });
+
+        // Connect filter to the card
+        cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+          cards: [
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/dashboard-filters/reproductions/15689.cy.spec.js:32]: Same suggestion as above of a possible function.

File: frontend/test/metabase/scenarios/dashboard-filters/reproductions/15695.cy.spec.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      has_field_values: "list",
+    });
+
+    cy.createQuestionAndDashboard({ questionDetails }).then(
+      ({ body: { id, card_id, dashboard_id } }) => {
+        cy.addFilterToDashboard({ filter, dashboard_id });
+
+        // Connect filter to the card
+        cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+          cards: [
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/dashboard-filters/reproductions/15695.cy.spec.js:37]: Same.

File: frontend/test/metabase/scenarios/dashboard-filters/reproductions/16103.cy.spec.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    cy.signInAsAdmin();
+  });
+
+  it("filter dropdown should not send request for values every time the widget is opened (metabase#16103)", () => {
+    // In this test we're using already present dashboard ("Orders in a dashboard")
+    cy.addFilterToDashboard({ filter, dashboard_id: 1 });
+
+    cy.log("Connect filter to the existing card");
+    cy.request("PUT", "/api/dashboard/1/cards", {
+      cards: [
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/dashboard-filters/reproductions/16103.cy.spec.js:32]: Same.

File: frontend/test/metabase/scenarios/dashboard-filters/reproductions/6660.cy.spec.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    cy.findByText("Days").click();
+    // Hours should appear in the selection box (don't click it)
+    cy.findByText("Hours");
+    // Minutes should appear in the selection box; click it
+    cy.findByText("Minutes").click();
+    // also check the "Include this minute" checkbox
+    // which is actually "Include" followed by "this minute" wrapped in <strong>, so has to be clicked this way
+    cy.contains("Include this minute").click();
+    // make sure the checkbox was checked
+    cy.findByRole("checkbox").should("be.checked");
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/dashboard-filters/reproductions/6660.cy.spec.js:34]: Here we seem to be falling back into procedural programming.

If there's any possible grouping of these acts, they could be good candidates to become functions.

Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/dashboard-filters/reproductions/6660.cy.spec.js:34]: This was just copy/paste.
This particular repro will most likely become obsolete by the extensive test coverage and therefore removed.

File: frontend/test/metabase/scenarios/dashboard-filters/reproductions/13150.cy.spec.js (hunk 2 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/dashboard-filters/reproductions/13150.cy.spec.js:45]: I hear you.
https://github.com/metabase/metabase/pull/16970#discussion_r667336235

These things take time, but are not forgotten.

**Review by gusaiani — APPROVED** [review]

File: frontend/test/metabase/scenarios/dashboard-filters/reproductions/6660.cy.spec.js (hunk 8 above)
Comment (gusaiani) [reply frontend/test/metabase/scenarios/dashboard-filters/reproductions/6660.cy.spec.js:34]: Gotcha 👍 Approved.

Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/dashboard-filters/reproductions/6660.cy.spec.js:34]: Thank you :)

**Pull request merged**